coverage.py

```diff
@@ -772,7 +772,7 @@ def _data_source_for_registration(cls, _db, data_source, autocreate=False):
             return DataSource.lookup(_db, cls.DATA_SOURCE_NAME)
         if isinstance(data_source, DataSource):
             return data_source
-        if isinstance(data_source, basestring):
+        if isinstance(data_source, str):
```

Comment:
The only way a bytestring data source might come in here is if it's part of an HTTP request, e.g. when someone is setting up a new OPDS import. I don't know whether we automatically decode form submissions as they come in.

I'm fine with replacing `isinstance(x, basestring)` with `isinstance(x, (bytes, str))` anywhere to avoid this situation.

Reply:
Yeah, I'll make that update replacing `isinstance(x, basestring)` to `isinstance(x, (bytes, str))`.